apps/api/src/domain/models/song/song.entity.ts

```diff
@@ -0,0 +1,102 @@
+import {
+    IsNonNegativeFiniteNumber,
+    IsOptional,
+    IsStringWithNonzeroLength,
+} from '@coscrad/validation';
+import { InternalError } from '../../../lib/errors/InternalError';
+import cloneToPlainObject from '../../../lib/utilities/cloneToPlainObject';
+import isNumberWithinRange from '../../../lib/validation/geometry/isNumberWithinRange';
+import { DTO } from '../../../types/DTO';
+import InconsistentTimeRangeError from '../../domainModelValidators/errors/context/invalidContextStateErrors/timeRangeContext/InconsistentTimeRangeError';
+import { Valid } from '../../domainModelValidators/Valid';
+import { resourceTypes } from '../../types/resourceTypes';
+import { TimeRangeContext } from '../context/time-range-context/time-range-context.entity';
+import { Resource } from '../resource.entity';
+
+type ContributorAndRole = {
+    contributorId: string;
+    role: string;
+};
+
+export class Song extends Resource {
+    readonly type = resourceTypes.song;
+
+    @IsOptional()
+    @IsStringWithNonzeroLength()
+    readonly title?: string;
+
+    @IsOptional()
+    @IsStringWithNonzeroLength()
+    readonly titleEnglish?: string;
+
+    // TODO Validate type
```

Comment:
Add a pivotal story and link it here.

Reply:
No story needed, added a nested validation rule.